content/asm_2.md

```diff
@@ -1,235 +1,508 @@
 
 ```assembly
-;; compare rax with 50
+;; compare value of the rax register with 50
 cmp rax, 50
 ```
 
-The `cmp` instruction just compares 2 values, but doesn't affect them and doesn't execute anything depend on result of comparison. For performing any actions after comparison there is conditional jump instructions. It can be one of it:
+The `cmp` instruction executes only comparison of its parameters without affecting values of the parameters themselves. To perform any actions after the comparison, there is conditional jump instructions. The list of these instructions:
 
-*  `JE` - if equal
-*  `JZ` - if zero
-*  `JNE` - if not equal
-*  `JNZ` - if not zero
-*  `JG` - if first operand is greater than second
-*  `JGE` - if first operand is greater or equal to second
-*  `JA` - the same that JG, but performs unsigned comparison
-*  `JAE` - the same that JGE, but performs unsigned comparison
+-  `JE`  - Jump if the values are equal.
+-  `JNE` - Jump if the values are not equal.
+-  `JZ`  - Jump if the difference between the two values is zero.
+-  `JNZ` - Jump if the difference between the two values is not zero.
+-  `JG`  - Jump if the first value is greater than the second.
+-  `JGE` - Jump if the first is greater or equal to the second.
+-  `JA`  - The same that JG, but performs unsigned comparison.
+-  `JAE` - The same that JGE, but performs unsigned comparison.
 
-For example if we want to make something like if/else statement in C:
+For example if we want translate something like this if/else statement written in C:
 
 ```C
 if (rax != 50) {
-    exit();
+    foo();
 } else {
-    right();
+    bar();
 }
 ```
 
-will be in assembly:
+to assembly, it will something like this:
 
 ```assembly
 ;; compare rax with 50
 cmp rax, 50
-;; perform .exit if rax is not equal 50
-jne .exit
-jmp .right
+;; jump to the label `.foo` if the value of the `rax` register is not equal to 50
+jne .foo
+;; jump to the label `.bar` otherwise
+jmp .bar
 ```
 
-There is also unconditional jump with syntax:
+In addition we can jump on a label without any conditions with the `jmp` instruction:
 
 ```assembly
-JMP label
+jmp .label
 ```
 
-For example:
+Often the unconditional jumps are used to simulate a loop. For example we have label and some code after it. This code executes anything, than we have condition and jump to the start of this code if condition is not successfully. The loops will be covered in next parts.
 
-```assembly
-_start:
-    ;; ....
-    ;; do something and jump to .exit label
-    ;; ....
-    jmp .exit
+### Program example
 
-.exit:
-    mov    rax, 60
-    mov    rdi, 0
-    syscall
-```
+Since we learned some basic arithmetic and control flow instructions, we can write our example. Before we will take a look at the program source code, I will remind that we are going to write a simple program that will calculate the sum of two integer numbers and if the sum equal to the third predefined number we will print a string. Otherwise just exist.
```

Comment:
```suggestion
Now that we've covered basic arithmetic and control flow instructions, we can write a simple program. As a reminder, this program will calculate the sum of two integers and compare it to a predefined third number. If the sum equals the third number, a string will be printed; otherwise, the program will just exit.
```